src/utils/Cargo.toml

```diff
@@ -10,3 +10,7 @@ libc = ">=0.2.39"
 vmm-sys-util = ">=0.2.1"
 
 net_gen = { path = "../net_gen" }
+
+[dev-dependencies]
```

Comment:
This was the previous way of testing the Tap. We removed the dependencies on pnet on purpose. See PR: https://github.com/firecracker-microvm/firecracker/pull/1238

Reply:
Ok, thanks ! I will try to find a solution in order to avoid the pnet dependency.

Reply:
The problem here is the fact that we need the frame injection logic both in the unit tests for the `Tap` and in the unit tests for the `virtio net` device. But the logic defnied inside `mod tests` can't be exported across crates. So we would have to duplicate this logic. If we use pnet, there are only a couple of lines of code, so duplicating them wouldn't be a big issue. If we don't use pnet there would be more than 100 lines of code that we would need to duplicate and this would be more problematic.

For the moment the `Tap` struct is only used inside the net emulation code. So I propose moving the tap inside the devices crate. This way the entire crate will be able to access the logic defined inside the `Tap` unit tests. @firecracker-microvm/compute-capsule  does anyone have any concern related to this approach ?

Reply:
I would also add that the current `Tap` implementation is very virtio-net specific at the moment as we inject the vnet information in it. It would need some changes to be truly library worth it.